### ISSUE TITLE
Gnome 41.2: wrong windows counter indicators count

### ISSUE THREAD

**Issue by ask6483** [micheleg/dash-to-dock#1629]
As seen on screenshot attached bellow windows counter indicators numbers are totally wrong:
- 3 opened nautilus (files) windows -> 4 indicators
- 1 STM32CubeIDE -> 4 indicators
- 1 Thunderbird window -> 2 indicators

Running gnome-shell over X server and reloading session doesn't help.

OS: Debian GNU/Linux bookworm/sid (testing)
Gnome 41.2

![image](https://user-images.githubusercontent.com/72493041/148085343-d00ee49a-0214-4cc4-8a64-1e2ac2c7fd11.png)

**Comment by mbiebl** [comment]
I can reproduce this as well. Disabling the extension and running the GNOME Shell builtin overview, I see the correct number for the windows counters.

**Comment by inthreedee** [comment]
I've noticed it has started picking up tooltips in the window count, which may be related to this.  Perhaps things like tooltips are confusing the count and causing it to get stuck at a wrong count

To reproduce, I'll use Firefox as an example
1. Open a site in a tab
2. Hover the mouse over the tab until the tooltip appears, showing the full title of the website
3. Dash to dock counts an extra window for Firefox
4. Move the mouse away from the tab and the count returns to what it was previously

**Comment by GeoffreyCoulaud** [comment]
I confirm that the issue is reproducible, I have the same problem, it's quite annoying.

System : Manjaro (unstable branch)
Gnome version : 41.3

**Comment by kanakshilledar** [comment]
I can also confirm the issue of wrong windows counter.
System: Endeavour OS Atlantis
Gnome version: 41.3

When browsing on firefox the window counter suddenly increases even if i have only one window present. I was able to get this issue in Nautilus when opening two different nautilus windows and copying one file from the other the window counter increments by 1.

**Comment by GeoffreyCoulaud** [comment]
If you want to speed up the process, test MR #1637 and give feedback on it.

**Comment by jesusignazio** [comment]
Same in GNOME Shell 41.3 on Archlinux.

**Comment by 3v1n0** [comment]
Can you provide some more infos of the windows causing the issue by using this in looking glass (from `alt+f2` → `lg`):

`Main.extensionManager._extensions.get('dash-to-dock@micxgx.gmail.com').imports.locations.getFileManagerApp().get_windows().map(w => [w.title, w.skip_taskbar,'|'])`

Or more generically, if you launch

`Main.extensionManager._extensions.get('dash-to-dock@micxgx.gmail.com').imports.locations.getFileManagerApp().get_windows()` and then you can inspect the windows by clicking on the list elements. Checking what's making them different from others.

**Comment by 3v1n0** [comment]
Also, related to #1644 ?

**Comment by kanakshilledar** [comment]
I did that `alt-f2` + `lg` and then typed this 
```
Main.extensionManager._extensions.get('dash-to-dock@micxgx.gmail.com').imports.locations.getFileManagerApp().get_windows().map(w => [w.title, w.skip_taskbar,'|'])
```
![got this output](https://user-images.githubusercontent.com/56501211/150463284-87971b6d-a870-424e-9e32-1438c63ae0ce.png)

for the next one
```
Main.extensionManager._extensions.get('dash-to-dock@micxgx.gmail.com').imports.locations.getFileManagerApp().get_windows()
```

![got this one](https://user-images.githubusercontent.com/56501211/150463464-50d94a83-2b3d-4c2b-be6c-06b6b853ddfb.png)


The issue is not only specific to nautilus but even after a long session of Firefox it's counter also increments by 1 but it happens randomly. I am able to reproduce that  by dragging one tab out of the window.

**Comment by 3v1n0** [comment]
Ok thanks... So, the #1637 PR idea is fine, but I guess only it works if you enabled isolated workspaces.

**Comment by kanakshilledar** [comment]
Do I need to do something else? Sorry I am new at this this is my first time participating in an github issue.

**Comment by 3v1n0** [comment]
No @kanakshilledar, thank you for the provided infos, we will handle them in #1637.

**Comment by GeoffreyCoulaud** [comment]
@3v1n0 since the PR is merged the issue could be closed ?

**Comment by 3v1n0** [comment]
Closed by be062eb

**Comment by dsager** [comment]
Thanks for fixing this! Any idea when there's going to be a new release including this?
